finetune/adapter.py

```diff
@@ -93,8 +93,9 @@ def main(fabric: L.Fabric, data_dir: Path, checkpoint_dir: Path, out_dir: Path):
     config = Config.from_name(name=checkpoint_dir.name)
     checkpoint_path = checkpoint_dir / "lit_model.pth"
     fabric.print(f"Loading model {str(checkpoint_path)!r} with {config.__dict__}")
-    with fabric.init_module():
+    with fabric.init_module(empty_init=False):
         model = GPT(config)
+        model.apply(model._init_weights)
```

Comment:
We need to initialize the model for the added adapter/lora weights, even though most will be replaced by the checkpoint.

We could consider just initializing those if we find that optimization useful